### PR TITLE
fixes #8998 - remove unnecessary routes

### PR DESCRIPTION
**Pull request by thomasmckay**
Work in progress but suggestions on better way welcome!

Goal is to remove all routes not allowed in SAM, both UI and API.

### CODE REVIEW

File: config/routes/overrides.rb [hunk 1]
```diff
@@ -13,4 +13,26 @@
 Foreman::Application.routes.draw do
   root :to => 'katello/dashboard#index'
   match '/dashboard', :to => 'katello/dashboard#index'
+
+  # foreman
+  match '*all/architectures', :to => proc { [404, {}, ['']] }
+  match '*all/architectures/*all', :to => proc { [404, {}, ['']] }
```
Comment (komidore64) [inline config/routes/overrides.rb]: i think you can accomplish with one line what you're attempting to do with lines 18-19

```ruby
match '*pre/architectures(/*post)', :to => proc { [404, {} ['']] }
```

Comment (thomasmckay) [inline config/routes/overrides.rb]: That works, updated

**Comment by komidore64** [comment]
ack

File: config/routes/overrides.rb [hunk 2]
```diff
@@ -13,4 +13,40 @@
 Foreman::Application.routes.draw do
   root :to => 'katello/dashboard#index'
   match '/dashboard', :to => 'katello/dashboard#index'
+
+  # foreman
+  match 'architectures(/*post)', :to => proc { [404, {}, ['']] }
```
Comment (komidore64) [inline config/routes/overrides.rb]: wonder if it's possible to have a conditional piece on the front...

Comment (thomasmckay) [inline config/routes/overrides.rb]: I did try several variations without success.

Comment (komidore64) [inline config/routes/overrides.rb]: cool cool. just wondering

**Pull request merged**